src/Microsoft.AspNetCore.SignalR.Client/HubConnectionFactory.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.Sockets;
+using Microsoft.AspNetCore.Sockets.Client;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public static class HubConnectionFactory
+    {
+        public static HubConnection Create(Uri url, ILoggerFactory loggerFactory = null)
```

Comment:
Don't we generally avoid default param values?

Reply:
we generally do. There is too many combinations here so I just short circuit this. Once we decide what we want to do we can decide if we want all the overloads or nullable logger (unless we do something different)